### ISSUE TITLE
5.12.33 Elasticsearch Performance: CPU utiliczation 100%

### ISSUE THREAD

**Issue by lesley-tw** [LuatixHQ/opencti#6851]
## Prerequisites

- [x] I read the [Deployment and Setup](https://docs.opencti.io/latest/deployment/overview) section of the OpenCTI documentation as well as the [Troubleshooting](https://docs.opencti.io/latest/deployment/troubleshooting) page and didn't find anything relevant to my problem.
- [x] I went through old GitHub issues and couldn't find anything relevant
- [x] I googled the issue and didn't find anything relevant

## Description

We recently upgraded from version 5.3.17 to 5.12.33 and have encountered a performance issue with the new version. For instance, when accessing the same campaign attack pattern page in the knowledge tab, the number of queries sent to Elasticsearch increases dramatically, despite using the same GraphQL query as before. Specifically, the new version triggers about 5,800 queries to Elasticsearch, causing the CPU usage to spike to 100% for just one page. In contrast, the old version only generated about 45 queries for the same page. We suspect that other pages might have similar issues, as even a reduced number of users is leading to the same CPU overload.

* Not sure it's the same issue with [issue#5029](https://github.com/OpenCTI-Platform/opencti/issues/5029)

## Environment

1. OS (where OpenCTI server runs): AWS EKS, 4 pods to serve this service, each pod 6 core cpu & 16 G memory
2. OpenCTI version: `5.12.33`
3. OpenCTI client: frontend or python
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:

1. Visit the same campaign, in knowledge tab, click attack pattern in the right menu.
![image](https://github.com/OpenCTI-Platform/opencti/assets/9364352/62bd7f0a-4895-47b0-a6c2-8afc9bcf1834)

In `5.3.17` 

- show 896 attack patterns in this page
- spend time 
![Screenshot 2024-04-30 at 4 05 47 PM](https://github.com/OpenCTI-Platform/opencti/assets/9364352/d866187e-eafc-42eb-853e-d59067191376)
    
- request payload
![Screenshot 2024-04-30 at 3 39 49 PM](https://github.com/OpenCTI-Platform/opencti/assets/9364352/bce72f22-66fe-434d-a547-fc473306ab9f)
    
- Elasticsearch query count: 45
![Screenshot 2024-04-30 at 1 59 43 PM](https://github.com/OpenCTI-Platform/opencti/assets/9364352/8214a69f-74d9-4a36-a692-8e2272306296)

    

In `5.12.33`

- show 896 attack patterns in this page
- spend time
![Screenshot 2024-04-30 at 4 06 06 PM](https://github.com/OpenCTI-Platform/opencti/assets/9364352/2b665e21-94ac-42f1-928b-d71e490603ef)
    
- request payload
![Screenshot 2024-04-30 at 4 14 48 PM](https://github.com/OpenCTI-Platform/opencti/assets/9364352/afd03f40-a00a-4809-b13c-99d1cb5930a9)

    
- Elasticsearch query count: 5,873
![Screenshot 2024-04-30 at 2 00 02 PM](https://github.com/OpenCTI-Platform/opencti/assets/9364352/685e7b4f-1e3d-4655-99a1-947e3b8dc816)

---
## Additional Information
In `5.3.17`:

* volume of data
![Screenshot 2024-04-30 at 2 41 28 PM (2)](https://github.com/OpenCTI-Platform/opencti/assets/9364352/313be847-359c-4517-881f-47f4f92b8155)

* Elasticsearch resource
   * 4 data node (each node: 4 core CPU & 32 G memory)
   * 2 master node (each node: 1 core CPU & 24 G memory)

In `5.12.33`:
* volume of data
![Screenshot 2024-04-30 at 2 36 51 PM (2)](https://github.com/OpenCTI-Platform/opencti/assets/9364352/5012830d-3783-4ae1-ab60-dfdc8efc8def)

* Elasticsearch resource
   * 10 data node (each node: 15 core CPU & 36 G memory)
   * 2 master node (each node: 2 core CPU & 24 G memory)

**Comment by richard-julien** [comment]
Seems related, this screen need to be improve with a specific api to load the matrix.

**Comment by vncloudsco** [comment]
We also had the same situation where we ran out of cpu resources for Elasticsearch

**Comment by trend-tanya-wang** [comment]
Hi @SamuelHassine, may I follow up this issue? Will OpenCTI fix this issue or any suggestions? Thank you.

**Comment by richard-julien** [comment]
Yes, will be handled by https://github.com/OpenCTI-Platform/opencti/issues/6662

**Comment by lesley-tw** [comment]
Hi @SamuelHassine
We are unsure if other pages have the same issue. Could you help check them?
Sometimes just reading a campaign with many relationships or listing reports might also cause elasticsearch CPU issues too.

**Comment by richard-julien** [comment]
if you have issue when other screen please create a specific ticket describing it. Thanks

**Comment by alice-debra** [comment]
Hello @lesley-tw,
https://github.com/OpenCTI-Platform/opencti/issues/6662 has handled this. If you’re still having any issues, feel free to reopen the issue.
We sincerely thank you for your feedback. 
Kind regards,
Alice
